lib/driver.js

```diff
@@ -484,6 +482,37 @@ export class XCUITestDriver extends BaseDriver {
     }
   }
 
+  async handleMjpegOptions() {
+    const mjpegServerPort = this.opts.mjpegServerPort || DEFAULT_MJPEG_SERVER_PORT;
+    await this.allocateMjpegServerPort(this.opts.udid,
+      mjpegServerPort, this.isRealDevice() ? true : false);
+    // turn on mjpeg stream reading if requested
+    if (this.opts.mjpegScreenshotUrl) {
+      this.log.info(`Starting MJPEG stream reading URL: '${this.opts.mjpegScreenshotUrl}'`);
+      this.mjpegStream = new mjpeg.MJpegStream(this.opts.mjpegScreenshotUrl);
+      await this.mjpegStream.start();
+    }
+  }
+
+  async allocateMjpegServerPort(udid, mjpegServerPort, usePortForwarding) {
+    this.log.debug(
+      `MJPEG broadcasting requested, forwarding MJPEG server port ${mjpegServerPort} ` +
```

Comment:
this log line is confusing as the method is executed even if we don't explicitly request the port forward